### PR TITLE
Cipher 1.9x speedup

### PR DESCRIPTION
**Pull request by klauspost**
Simplify cipher loop and unroll once.

```
λ benchcmp before.txt after.txt | grep -v "naive"
benchmark                                          old ns/op     new ns/op     delta
BenchmarkCipher/bytes=7;offset=1-32                7.17          7.79          +8.65%
BenchmarkCipher/bytes=125;offset=0-32              24.5          23.1          -6.07%
BenchmarkCipher/bytes=1024;offset=0-32             126           63.4          -49.87%
BenchmarkCipher/bytes=4096;offset=0-32             462           244           -47.26%
BenchmarkCipher/bytes=4100;offset=4-32             460           249           -45.93%
BenchmarkCipher/bytes=4099;offset=3-32             463           250           -45.93%
BenchmarkCipher/bytes=32775;offset=49-32           3619          1936          -46.50%

benchmark                                          old MB/s     new MB/s     speedup
BenchmarkCipher/bytes=7;offset=1-32                976.74       898.93       0.92x
BenchmarkCipher/bytes=125;offset=0-32              5092.84      5423.57      1.06x
BenchmarkCipher/bytes=1024;offset=0-32             8103.00      16159.29     1.99x
BenchmarkCipher/bytes=4096;offset=0-32             8870.86      16818.08     1.90x
BenchmarkCipher/bytes=4100;offset=4-32             8917.63      16491.32     1.85x
BenchmarkCipher/bytes=4099;offset=3-32             8854.38      16379.58     1.85x
BenchmarkCipher/bytes=32775;offset=49-32           9056.37      16926.83     1.87x
```

I tried a few variations, but this seemed fine without too many changes.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-latest, macos-latest, windows-latest ]
-        go: [ '1.15', 'stable', 'oldstable' ]
+        go: [ '1.20.x', '1.21.x', '1.22.x' ]
```
[the remark this answers is not in the document]
Comment (klauspost) [reply .github/workflows/main.yml:20]: Dunno. Seemed like you wanted some older versions to be tested. I just wanted CI to build ;)

File: cipher.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func Cipher(payload []byte, mask [4]byte, offset int) {
 	ln := remain[mpos]
 	// Count number of bytes will processed one by one from the end of payload.
 	// This is done to process payload by 8 bytes in each iteration of main loop.
-	rn := (n - ln) % 8
+	rn := (n - ln) % 16
```
[the remark this answers is not in the document]
Comment (klauspost) [reply cipher.go:29]: Correct. I see the comment still references 8. I will fix.

File: cipher.go [hunk 3]
```diff
@@ -44,15 +44,15 @@ func Cipher(payload []byte, mask [4]byte, offset int) {
 	)
 	// Skip already processed right part.
 	// Get number of uint64 parts remaining to process.
-	n = (n - ln - rn) >> 3
+	n = (n - ln - rn) >> 4
```
[the remark this answers is not in the document]
Comment (klauspost) [reply cipher.go:47]: Yeah `/16` instead of `/8`.

**Comment by cristaloleg** [comment]
Yeah, it's:(

File: .github/workflows/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (cristaloleg) [reply .github/workflows/main.yml:20]: Can we stick to `stable` and `oldstable` please?

I've merged https://github.com/gobwas/ws/pull/191 can you rebase please?

[the remark this answers is not in the document]
Comment (klauspost) [reply .github/workflows/main.yml]: done

**Review by cristaloleg — APPROVED** [review]
Great, thanks!

**Pull request merged**